program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,61 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from django.db.models import F, Window
+from django.db.models.functions import RowNumber
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+
+
+
+
+# TODO Do I add an @schema thing here?
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent CERTAIN intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK
+     * 400: Query returned no results
+    """
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    def get_queryset(self):
```

Comment:
I'm not an expert on Django Rest Framework, but I think that `get_queryset` is a method you'd override on a DRF `ViewSet` or other generic view, not the `APIView`. I'd think you would implement the `APIView.get()` method here. Or, you could use `GenericAPIView` or `ListAPIView`.

The logic here is pretty standard for a list endpoint, so using a generic would reduce the boilerplate you have after you define the queryset.

Reply:
I used the APIView.get() 👍